client/core/trade.go

```diff
@@ -1276,6 +1282,91 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) (re
 	return false, false
 }
 
+// updateDynamicSwapOrRedemptionFeesPaid updates the fees used for dynamic fee
+// checker transactions. We do not know the exact fees the tx will use until
+// they are mined, so this waits until they are mined and updates the value for
+// the entire trade.
+//
+// TODO: This method does not persist if the client is shut down. It probably
+// should because payed fees will be incorrect forever for this trade if not.
+func (t *trackedTrade) updateDynamicSwapOrRedemptionFeesPaid(ctx context.Context, match *matchTracker, isInit bool) {
+	wallet := t.wallets.fromWallet
+	if !isInit {
+		wallet = t.wallets.toWallet
+	}
+	feeChecker, dynamic := wallet.Wallet.(asset.DynamicSwapOrRedemptionFeeChecker)
+	if !dynamic {
+		return
+	}
+	stopChecks := func() {
+		if isInit {
+			match.needsDynamicSwapFeeConfirms = false
+		} else {
+			match.needsDynamicRedemptionFeeConfirms = false
+		}
+	}
+	txType := "swap"
+	if !isInit {
+		txType = "redemption"
+	}
+	var coinID, contractData []byte
+	// Check if a swap or redeem coin id has been populated in the
+	// match tracker. If it has we ask the wallet for the fees paid
+	// and add that to either the total swap or redeem fees for the
+	// trade.
+	if isInit {
+		coinID = []byte(match.MetaData.Proof.MakerSwap)
+		if match.Side != order.Maker {
+			coinID = []byte(match.MetaData.Proof.TakerSwap)
+		}
+		contractData = match.MetaData.Proof.ContractData
+	} else {
+		coinID = []byte(match.MetaData.Proof.MakerRedeem)
+		if match.Side != order.Maker {
+			coinID = []byte(match.MetaData.Proof.TakerRedeem)
+		}
+		contractData = match.MetaData.Proof.CounterContract
+	}
+	secretHash := match.MetaData.Proof.SecretHash
+	if coinID == nil {
+		// If there is no coin ID yet and the match was revoked, assume
+		// the transaction will never happen.
+		if match.MetaData.Proof.IsRevoked() {
+			stopChecks()
+		}
+		return
+	}
+	checkFees := feeChecker.DynamicSwapFeesPaid
+	if !isInit {
+		checkFees = feeChecker.DynamicRedemptionFeesPaid
+	}
+	actualSwapFees, secrets, err := checkFees(ctx, coinID, contractData)
+	if err != nil {
+		if errors.Is(err, asset.CoinNotFoundError) || errors.Is(err, asset.ErrNotEnoughConfirms) {
+			return
+		}
+		t.dc.log.Errorf("Failed to determine actual %s transaction fees paid for "+
+			"match %s: %v", txType, match, err)
+		stopChecks()
+		return
+	}
+	// Only add the tx fee once.
+	if !bytes.Equal(secrets[0], secretHash) {
+		stopChecks()
+		return
+	}
+	if isInit {
+		t.metaData.SwapFeesPaid += actualSwapFees
+	} else {
+		t.metaData.RedemptionFeesPaid += actualSwapFees
+	}
+	err = t.db.UpdateOrderMetaData(t.ID(), t.metaData)
+	if err != nil {
+		t.dc.log.Errorf("Error updating order metadata for order %s: %v", t.ID(), err)
+	}
+	stopChecks()
```

Comment:
Ah at the end we need to do:

```go
note := newMatchNote(TopicConfirms, "", "", db.Data, t, match)
t.notify(note)
```

Otherwise we have to wait for `(*trackedTrade) confirmRedemption` to send a `MatchNote` with the updated fees.

Reply:
Notifying now anyway.